### PR TITLE
feat-4489-transcribe-audio-into-written-text

### PR DESCRIPTION
**Pull request by Yordaniss**
Issue #4489 Write a deepgramTranscribeAudio() Function using PHP

Closes https://github.com/appwrite/appwrite/issues/4489

* created function to transcribe audio into written text
* added guzzle client dependency
* added vendor to gitignore
* added readme file

**Examples:**

Success:
Bildschirmfoto 2022-11-08 um 22 35 06

Error:
Bildschirmfoto 2022-11-08 um 22 37 33

### CODE REVIEW

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by Yordaniss** [comment]
Hello @Meldiron . Awesome, thank you for the feedback and sure i will stay in touch :)

File: php/deepgram_transcribe_audio/index.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+require 'vendor/autoload.php';
+
+use GuzzleHttp\Client;
+
+return function($req, $res) {
+    try {
+        $payload = \json_decode($req['payload'], true);
+        if (!isset($payload['fileUrl']) || empty($fileUrl = \trim($payload['fileUrl']))) {
+            throw new \Exception('Please provide valid file url.');
```
Comment (stnguyen90) [inline php/deepgram_transcribe_audio/index.php:11]: Please make sure not to throw exceptions. and return an error response like:

```json
{"success":false,"message":"Please provide a valid file URL."}
```

instead.

File: php/deepgram_transcribe_audio/index.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            empty($deepgramSecretApiKey = \trim($payload['variables']['deepgramApiSecretKey']))
+        ) {
+            throw new \Exception('Please provide valid deepgram api secret key under variables.');
+        }
+
+        $extension = pathinfo($fileUrl, PATHINFO_EXTENSION);
+
+        if ($extension !== 'wav') {
+            throw new \Exception('Please provide valid file url extension. This should be wav.');
+        }
```
Comment (stnguyen90) [inline php/deepgram_transcribe_audio/index.php:24]: Deepgrams supports more than just wav files don't they?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

File: php/deepgram_transcribe_audio/index.php (hunk 2 above)
Comment (Yordaniss) [reply php/deepgram_transcribe_audio/index.php:24]: I thought, it's should be only .wav format like in Issue task described. [Issue](https://github.com/appwrite/appwrite/issues/4489)

"The payload must accept a WAV file URL, and the response from Deepgramm should be returned."

If I misunderstood, no problem i can remove this condition

**Comment by Yordaniss** [comment]
Hello @stnguyen90  thanks a lot for feedback. I did all requested changes until ".wav" format comment. I the changes still requested, please let me know.

File: php/deepgram_transcribe_audio/index.php (hunk 2 above)
Comment (stnguyen90) [reply php/deepgram_transcribe_audio/index.php]: ok sure. fair enough

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Comment by Yordaniss** [comment]
Great! Glad to help 🙏

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**